src/plugins/intel_gpu/src/kernel_selector/kernels/resample/resample_kernel_onnx.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "resample_kernel_onnx.h"
+#include <vector>
+#include <kernel_selector_utils.h>
+
+namespace kernel_selector {
+
+static constexpr size_t sub_group_size = 16;
+
+size_t ResampleKernelOnnx::GetOptimalBlockSize(const resample_params& params) const {
+    std::vector<size_t> block_width = { 16, 8, 4, 2, 1 };
+    for (auto& w : block_width) {
+        if (params.outputs[0].X().v % w == 0) {
+            return w;
+        }
+    }
+    return 1;
+}
+
+static size_t GetOptimalDivisor(const size_t input_size, size_t max_val = 16) {
+    for (size_t s = max_val; s > 0; --s) {
+        if (input_size % s == 0) {
+            return s;
+        }
+    }
+    return 1;
+}
+
+ParamsKey ResampleKernelOnnx::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableInputDataType(Datatype::UINT8);
+    k.EnableInputDataType(Datatype::INT8);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::UINT8);
+    k.EnableOutputDataType(Datatype::INT8);
+
+    // 5d formats
+    k.EnableInputLayout(DataLayout::b_fs_zyx_fsv16);
```

Comment:
What about bfzyx format? Will it work?

Reply:
it is supported by resample_ref, so it should work.